plugin/forwardcrd/README.md

```diff
@@ -0,0 +1,207 @@
+zone specified in the server's block. It will allow any `Forward` resource that
+matches or includes the zone as a suffix. If **ZONES** are specified it allows
+any zone listed as a suffix.
+
+```
+forwardcrd [ZONES...] {
+  endpoint URL
+  tls CERT KEY CACERT
+  kubeconfig KUBECONFIG [CONTEXT]
+  namespace [NAMESPACE]
+}
+```
+
+* `endpoint` specifies the **URL** for a remote k8s API endpoint.  If omitted,
+  it will connect to k8s in-cluster using the cluster service account.
+* `tls` **CERT** **KEY** **CACERT** are the TLS cert, key and the CA cert file
+  names for remote k8s connection.  This option is ignored if connecting
+  in-cluster (i.e. endpoint is not specified).
+* `kubeconfig` **KUBECONFIG [CONTEXT]** authenticates the connection to a remote
+  k8s cluster using a kubeconfig file.  **[CONTEXT]** is optional, if not set,
+  then the current context specified in kubeconfig will be used.  It supports
+  TLS, username and password, or token-based authentication.  This option is
+  ignored if connecting in-cluster (i.e., the endpoint is not specified).
+* `namespace` **[NAMESPACE]** only reads `Forward` resources from the namespace
+  listed. If this option is omitted then it will read from the default
+  namespace, `kube-system`. If this option is specified without any namespaces
+  listed it will read from all namespaces.  **Note**: It is recommended to limit
+  the namespace (e.g to `kube-system`) because this can be potentially misused.
+  It is ideal to keep the level of write access similar to the `coredns`
+  configmap in the `kube-system` namespace.
+
+## Ready
+
+This plugin reports readiness to the ready plugin. This will happen after it has
+synced to the Kubernetes API.
+
+## Ordering
+
+Forward behavior can be defined in three ways, via a Server Block, via the *forwardcrd* plugin, and via the *forward* plugin.  If more than one of these methods is employed and a query falls within the zone of more than one, CoreDNS selects which one to use based on the following precedence:
+Corefile Server Block -> *forwardcrd* plugin -> *forward* plugin.
+
+When `Forward` CRDs and Server Blocks define stub domains that are used,
+domains defined in the Corefile take precedence (in the event of zone overlap).
+e.g. if the
+domain `example.com` is defined in the Corefile as a stub domain, and a
+`Forward` CRD record defined for `sub.example.com`, then `sub.example.com` would
+get forwarded to the upstream defined in the Corefile, not the `Forward` CRD.
+
+When using *forwardcrd* and *forward* plugins in the same Server Block, `Forward` CRDs
+take precedence over the *forward* plugin defined in the same Server Block.
+e.g. if a `Forward` CRD is defined for `.`, then no queries would be
+forwarded to the upstream defined in the *forward* plugin of the same Server Block.
+
+## Metrics
+
+`Forward` CRD metrics are all labeled in a single zone (the zone of the enclosing
+Server Block).
+
+## Examples
```

Comment:
A real-world example of how forwardcrd would be used in an actual CoreDNS Corefile as deployed in a kubernetes cluster would be helpful.